gprofiler/profilers/perf.py

```diff
@@ -245,6 +247,14 @@ def __init__(
         self._node_processes_attached: List[Process] = []
         self._perf_memory_restart = perf_memory_restart
         switch_timeout_s = duration * 3  # allow gprofiler to be delayed up to 3 intervals before timing out.
+        extra_args = []
+        if is_profiler_disabled(perf_mode):
+            try:
+                extra_args.extend(
+                    perf_default_event_works(Path(self._profiler_state.storage_dir), self._profiler_state.stop_event)
+                )
+            except PerfNoSupportedEvent:
+                logger.critical("Failed to determine perf event to use")
```

Comment:
You should exit after this, no? raise a critical error that'll cause gprofiler to exit.

Reply:
how should we do that? can we just `exit(1)` when some profilers have already started?

Reply:
talked f2f, it's ok to `sys.exit(1)` here
although we could just let the exception propagate couldn't we?

Reply:
We can let the exception propagate. Please post on the PR an example log displaying how it'd look when this fail (you can inject a fault to the code to make sure it does break, for example remove the "ok" return)

Reply:
great, let's keep this unresolved so we'll remember to do it when I run this?

Reply:
I resolved but added a checkbox task in the PR desc